eng/common/pipelines/templates/steps/credscan.yml

```diff
@@ -9,7 +9,12 @@ steps:
     if ("$(Build.Reason)" -eq 'PullRequest') {
       $targetBranch = "origin/$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
       $changedFiles = git diff $targetBranch HEAD --name-only --diff-filter=d
-      $changedFiles | ForEach-Object { Add-Content -Path "${{ parameters.SourceDirectory }}/credscan.tsv" -Value "${{ parameters.SourceDirectory }}/$_"}
+      foreach ($filePath in $changedFiles){
+        if ($filePath-match [System.IO.Path]::GetInvalidPathChars()) {
```

Comment:
@sima-zhu how is it possible that we have invalid path chars in our files? If they are invalid I wouldn't expect us to be able to write the file to disk at all.

Reply:
I did see several path name is different than the ones checkin the github repo.
E.g.
<img width="718" alt="image" src="https://user-images.githubusercontent.com/48036328/153970739-f06d991e-4101-4f68-b29a-b63810594e19.png">

Reply:
The path name interpret differently in .tsv file.
<img width="1215" alt="image" src="https://user-images.githubusercontent.com/48036328/153970948-26c5ec29-13bf-482f-80da-466f0b00dd9a.png">

Reply:
However, the cred scan tool is not able to read the file with name like, .*(True).json